bittide/src/Data/Constraint/Nat/Extra.hs

```diff
@@ -135,3 +135,7 @@ minLeq = unsafeCoerce (Dict :: Dict (0 <= 0))
 -- | Postulates that the minimum of a and b can't be larger than b
 maxGeqPlus :: forall a b c. Dict (a <= Max a b + c)
 maxGeqPlus = unsafeCoerce (Dict :: Dict (0 <= 0))
+
+-- | Transitivity of @<=@.
+leqTrans :: forall n m k. (n <= m, m <= k) => Dict (n <= k)
```

Comment:
This function already exists in the `constraints`  package: https://hackage.haskell.org/package/constraints-0.14/docs/Data-Constraint-Nat.html#v:leTrans